packages/legend-graph/src/graph-manager/protocol/pure/v1/engine/V1_Engine.ts

```diff
@@ -698,17 +702,27 @@ export class V1_Engine {
   async runQuery(
     input: V1_ExecuteInput,
     options?: ExecutionOptions,
-  ): Promise<V1_ExecutionResult> {
+  ): Promise<{
+    executionResult: V1_ExecutionResult;
+    executionTraceID?: string;
+  }> {
     try {
-      const executionResultInText = await this.runQueryAndReturnString(
+      const executionResultMap = await this.runQueryAndReturnString(
         input,
         options,
       );
+      const executionResultInText =
+        executionResultMap.get(V1_EXECUTION_RESULT) ?? '';
       const rawExecutionResult =
         returnUndefOnError(() =>
           this.parseExecutionResults(executionResultInText, options),
         ) ?? executionResultInText;
-      return V1_serializeExecutionResult(rawExecutionResult);
+      const executionResult = V1_serializeExecutionResult(rawExecutionResult);
+      const traceID = executionResultMap.get(V1_ZIPKIN_TRACE_HEADER);
+      if (traceID) {
+        return { executionResult: executionResult, executionTraceID: traceID };
+      }
+      return { executionResult: executionResult };
```

Comment:
Nit: don't need to include value separately if it's the same as the property name:
```suggestion
        return { executionResult, executionTraceID: traceID };
      }
      return { executionResult };
```